### PR TITLE
masked_log_softmax -inf for masked values

### PR DESCRIPTION
**Pull request by MoisesHer**
## Description ##
This PR sets -INF as returned value in FWD for masked values in log-softmax. log(0) = -INF

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] Change returned values in FWD log-softmax for masked values. Before returning 0, now returning -INF
- [x] Remove duplicated feature: scale_factor & temperature

## Comments ##
In Backwards, the gradient for -INF elements should be +INF, but this would introduce numerical issues.
Instead, I propose to set those with zero.
Similar discussion for PyTorch can be found [here](https://github.com/pytorch/pytorch/issues/31829)

@sxjscience

### CODE REVIEW

**Comment by MoisesHer** [comment]
@mxnet-bot run ci [unix-cpu]

**Review by sxjscience — APPROVED** [review]
LGTM
